conn.go

```diff
@@ -36,11 +36,29 @@ func (pr *PacketResponse) ReadPacket() (*ber.Packet, error) {
 	return pr.Packet, pr.Error
 }
 
+type messageContext struct {
+	id        int64
+	done      chan struct{}
+	responses chan *PacketResponse
```

Comment:
and this should only be closed from `processMessages`, and only sent to from `sendResponse`